e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js

```diff
@@ -256,8 +284,42 @@ describe(
       });
 
       getDashboardCard().within(() => {
-        cy.findByText("0").should("be.visible");
+        cy.findByText("Sleep card").should("be.visible");
+      });
+    });
+
+    it("should preserve filter value when navigating between the dashboard and the question", () => {
+      createDashboardWithSlowCard();
+      cy.get("@dashboardId").then(id => {
+        // Explicitly not using `visitDashboard` helper here because it also
+        // sets an intecept for the same route `POST /api/dashboard/*/dashcard/*/card/*/query`
+        // that we're listening to and asserting on
+        cy.visit(`/dashboard/${id}`);
+        cy.wait("@dashcardQuery");
+        cy.get("@dashcardQuery.all").should("have.length", 1);
       });
+
+      filterWidget().findByPlaceholderText("sleep").type("5{enter}");
+
+      cy.wait("@dashcardQuery");
+      cy.get("@dashcardQuery.all").should("have.length", 2);
+
+      filterWidget().findByPlaceholderText("sleep").should("have.value", "5");
+      getDashboardCard().within(() => {
+        cy.findByText("Sleep card").click();
+        cy.wait("@card");
+      });
+
+      queryBuilderHeader().findByLabelText("Back to Sleep dashboard").click();
+
+      filterWidget().findByPlaceholderText("sleep").should("have.value", "5");
+      getDashboardCard().within(() => {
+        cy.findByText("Sleep card").should("be.visible");
+        cy.get(".cellData").should("be.visible").and("contain", 5);
+      });
+
+      cy.log("Dashcard data should not be re-requested");
+      cy.get("@dashcardQuery.all").should("have.length", 2);
```

Comment:
The assertion is the opposite of what landed in `master`: https://github.com/metabase/metabase/pull/32192/files#diff-ab4bbcb5d3e1ba388de75ed1dcad298ad17ab278ac588837b56ee59ac51a313aR314-R315

Is this intentional?


Also, it does not match the test name.